server/src/main/java/org/apache/iotdb/db/engine/compaction/level/LevelCompactionTsFileManagement.java

```diff
@@ -498,6 +498,7 @@ private void merge(List<List<TsFileResource>> mergeResources, boolean sequence,
         return;
       }
     }
+    isSeqMerging = true;
```

Comment:
there will be a deadlock problem here when ```isSeqMerging = true``` and unseq merge is called in line 509